dora/core/common/src/main/java/alluxio/membership/WorkerClusterSnapshot.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.membership;
+
+import alluxio.collections.IndexDefinition;
+import alluxio.collections.IndexedSet;
+import alluxio.wire.WorkerIdentity;
+import alluxio.wire.WorkerInfo;
+
+import com.google.common.collect.Iterators;
+
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Snapshot for a cluster view.
+ */
+public final class WorkerClusterSnapshot implements WorkerClusterView {
+
+  private final IndexedSet<WorkerInfo> mWorkers;
+
+  private static final IndexDefinition<WorkerInfo, WorkerIdentity> INDEX_WORKER_ID =
+      IndexDefinition.ofUnique(WorkerInfo::getIdentity);
+  private static final IndexDefinition<WorkerInfo, String> INDEX_STATE =
+      IndexDefinition.ofNonUnique(WorkerInfo::getState);
```

Comment:
who's using this index?

Reply:
Or I think it's also fair to just keep two sets, one for LIVE(usable) and one for LOST(unusable). It's an immutable view/snapshot anyways so there's no worry in updating those two sets (therefore needing a lock).

Reply:
then equality may be easier (even if later we find it hard to implement version/timestamp comparison). Equality will be
```
1. check two views have same live worker number
2. check two views have same total worker number (live+lost=total)
3. O(n) comparison
```
Most frequently we just add/remove workers, or workers start/stop, so check 1 and check 2 should cover majority of changes. Worker restarting with the same id but different address should be rare, or captured by check 1/2 already because a restart takes at least seconds so the membership manager will know?

Reply:
removed this index as the design has fundamentally changed.